templates/server-discovery-role.yaml

```diff
@@ -1,7 +1,8 @@
 {{ template "vault.mode" . }}
-{{- if ne .mode "external" }}
```

Comment:
I think we may want to leave the external checks in place for the discovery role and binding? IIRC the external mode was setup to only require setting `injector.externalVaultAddr` to disable all the server things in the chart (or at least most of them): https://developer.hashicorp.com/vault/docs/platform/k8s/helm/run#external-mode

Reply:
My thinking on this one was it's strictly a subset of the `{{- if eq .mode "ha" }}` check below, so unnecessary. But happy to put it back in if it might be wrong/risky.